docs/download.rst

```diff
@@ -173,15 +174,20 @@ Downloading API
 
    :query _refresh: use ``_refresh=1`` to force Tecken to update cache
 
+   :resheader Location: redirect location for the file
+   :resheader Content-Length: length in bytes for the file
+   :resheader Content-Encoding: (optional) content encoding for the file; note
+       that ``.sym`` files are compressed even thought he file extension doesn't
+       indicate that
```

Comment:
I think these are the critical response headers. I'm a little fuzzy on HTTP 302s, though. Does this sound right? Are there other headers we want to include here?

Reply:
Only the Location header is in the response from Tecken. The other headers are part of the response from S3. We should make that clear in this list.

The 302 response from Tecken will have these headers:

* Location – the redirect URL
* Debug-Time – the time if Debug:true was in the request. (I'm unsure we really want this to be part of the documented interface)
* Content-Length: 0 – a redirect response doesn't have a body. We don't need to document this.
* Content-Type: text/html – Django seems to be putting this there by default. It doesn't really make sense.
* There's no Content-Encoding header in the Tecken response. There's no content, so specifying an encoding would be pointless.

If we follow the redirect, we talk directly to S3. The response include these headers:

* Content-Encoding: gzip – If the file is compressed. In general, the file should only be returned with this content encoding if the client indicated that it can deal with it with Accept-Encoding: gzip. However, S3 will always return it if the file was uploaded with this encoding. (GCS actually decompresses the file on the server side if the client doesn't indicate Accept-Encoding: gzip.) A content encoding is only used for transport and should be transparent to the end user. You get the "actual" content after decoding. For example you can navigate to https://s3.us-west-2.amazonaws.com/org.mozilla.crash-stats.symbols-public/v1/xpcshell/4C4C44B655553144A1B2DAA18738D74B0/xpcshell.sym in Firefox, and you will see the text content. The file is still stored and transmitted compressed, but the Content-Encoding header indicates to Firefox that the file should be decompressed first. In curl you can use `curl --compressed` to get the same behaviour; curl will list all the content encodings it supports in the Accept-Encoding header, and if the response indicates that one of these encodings was used the content will be decoded accordingly. 
* Content-Length: The length of the response body. If the body is compressed, it's the size of the compressed body.
* Content-Type: The content type of the response after decompressing it. Will be text/plain for symbol files.

I'm not actually sure whether this response makes things clearer or more confusing, but I'll send it anyway. The important point is that we should make clear what headers are in Tecken's response, and what headers are in S3's response, if we want to list the latter as well.

Reply:
We also need to think about the subtle difference in behaviour between S3 and GCS. Maybe we can add the Accept-Encoding: gzip header at the load balancer level in GCS to make sure the behviour is the same, but I'll have to look into it.